x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/catch_function_not_found_error.ts

```diff
@@ -5,26 +5,81 @@
  * 2.0.
  */
 
-import { catchError, of, OperatorFunction, throwError } from 'rxjs';
+import { catchError, filter, of, OperatorFunction, share, throwError } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { MessageRole } from '../../../../common';
 import {
-  createFunctionLimitExceededError,
+  ChatCompletionChunkEvent,
   isFunctionNotFoundError,
-  MessageAddEvent,
+  MessageOrChatEvent,
+  StreamingChatResponseEventType,
 } from '../../../../common/conversation_complete';
-import { createServerSideFunctionResponseError } from '../../util/create_server_side_function_response_error';
+import { emitWithConcatenatedMessage } from '../../../../common/utils/emit_with_concatenated_message';
 
-export function catchFunctionNotFoundError<T>(): OperatorFunction<T, T | MessageAddEvent> {
-  return catchError((error) => {
-    if (isFunctionNotFoundError(error)) {
-      const functionLimitExceededError = createFunctionLimitExceededError();
-      return of(
-        createServerSideFunctionResponseError({
-          name: error.meta.name,
-          error: functionLimitExceededError,
-        })
-      );
-    }
+function appendErrorMessageToAssistantResponse(): OperatorFunction<
+  MessageOrChatEvent,
+  MessageOrChatEvent
+> {
+  return (source$) => {
+    return source$.pipe(
+      filter(
+        (msg): msg is ChatCompletionChunkEvent =>
+          msg.type === StreamingChatResponseEventType.ChatCompletionChunk
+      ),
+      emitWithConcatenatedMessage(async (concatenatedMessage) => {
+        return {
+          ...concatenatedMessage,
+          message: {
+            ...concatenatedMessage.message,
+            content: `${concatenatedMessage.message.content}\n\n${i18n.translate(
+              'xpack.observabilityAiAssistant.functionCallLimitExceeded',
+              {
+                defaultMessage:
+                  '\n\nNote: the Assistant tried to call a function, even though the limit was exceeded',
```

Comment:
Is this error message correct? Looks like this called when the function is not found

Reply:
Ah, I see it's called for the branch `if (functionLimitExceeded)`. 

Perhaps rename `appendErrorMessageToAssistantResponse`  to `appendFunctionLimitExceededErrorMessageToAssistantResponse `